src/core/data/grid/gridlayout.hpp

```diff
@@ -1323,7 +1323,14 @@ namespace core
          */
         NO_DISCARD std::uint32_t constexpr static nbrDualGhosts_()
         {
-            return (interp_order + 1) / 2 + nbrParticleGhosts();
+            if constexpr (((interp_order + 1) / 2 + nbrParticleGhosts()) % 2 != 0)
+            {
+                return (interp_order + 1) / 2 + nbrParticleGhosts() + 1;
+            }
+            else
+            {
+                return (interp_order + 1) / 2 + nbrParticleGhosts();
+            }
```

Comment:
could we not consider something which is a bit more obvious at a glance?

for instance

```
static_assert(interp_order > 0 and interp_order);
constexpr static  ghosts = std::array{2,2,4};
return ghosts[interp_order - 1]; 
```

Reply:
yes !
but also we need to write a decent comment to explain those magic numbers.